### ISSUE TITLE
Inference reporter callback

### ISSUE THREAD

**Issue by clumsy** [NVIDIA-NeMo/NeMo#14993]
**Is your feature request related to a problem? Please describe.**

Given an existing PyTorch Lightning training/validation recipe it should be possible to produce inference data the model produces, e.g. logprobs (and maybe later logits), input/output tokens (and the decoded version) to assess the quality and troubleshoot model training. This data can be used to compare to other inference runs.

Right now several script variants are provided and very limited documentation about how to use it.

**Describe the solution you'd like**

It should be possible to simply add a callback to generate all this data using `LightningModule` (model) and `LightningDataModule` (data). It should be generic enough to be plugged in as is.

**Describe alternatives you've considered**

Create scripts to mimic this for each model/setup individually.

**Additional context**

https://www.aizhuanlan.net/blog/detail/df4a1f21-ac29-48cc-8a53-5e6a6d48e298

**Comment by cuichenx** [comment]
Hi, our LLM/VLM work has migrated to [Megatron-Bridge](https://github.com/NVIDIA-NeMo/Megatron-Bridge/tree/main), where lightning is removed. You can now add this directly in the training loop yourself.

**Comment by clumsy** [comment]
But most of this logic is Megatron's inference-engine specific @cuichenx, I've found this part to be barely documented.

Any chance something generic can be added to Megatron-Bridge? Lightning plumbing can be left out to the end user, sure.

**Comment by gautham-kollu** [comment]
Megatron-bridge using modules defined in megatron-lm(megatron/core a submodule of megatron-lm).  
This appears to help with dumping logits  https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/models/gpt/gpt_model.py#L672-L682

Can you set `model.config_logger_dir=<>` and see if that helps ?

**Comment by pzelasko** [comment]
Closing this issue as "won't fix".
The following collections have been moved to separate repos in https://github.com/NVIDIA-NeMo organization: avlm, llm, multimodal, multimodal-autoregressive, vlm, speechlm, diffusion
Please move the discussion to the related repo if the issue persists.
